src/library/dao/playlistdao.cpp

```diff
@@ -227,6 +228,51 @@ void PlaylistDAO::deletePlaylist(const int playlistId) {
     }
 }
 
+int PlaylistDAO::deleteAllPlaylistsWithFewerItems(PlaylistDAO::HiddenType type, int length) {
+    QSqlQuery query(m_database);
+
+    ScopedTransaction transaction(m_database);
+    query.prepare(QStringLiteral(
+            "SELECT id FROM Playlists  "
+            "WHERE (SELECT count(playlist_id) FROM PlaylistTracks WHERE "
+            "Playlists.ID = PlaylistTracks.playlist_id) < :length AND "
+            "Playlists.hidden = :hidden"));
+    query.bindValue(":hidden", static_cast<int>(type));
+    query.bindValue(":length", length);
+    if (!query.exec()) {
+        LOG_FAILED_QUERY(query);
+        return -1;
+    }
+
+    QStringList idStringList;
+    while (query.next()) {
+        idStringList.append(query.value(0).toString());
+    }
+    QString idString = idStringList.join(",");
+
+    qDebug() << "Delete Playlists: " << idString;
```

Comment:
qInfo() << "Deleting" << idStringList.size() << "playlists of type" << type << "that contain fewer than" << minNumerOfTracks << "tracks";

The ids should not matter if the code has been tested thoroughly.